complex/ft_test.c

```diff
@@ -212,7 +242,7 @@ static void ft_format_iov_random(struct iovec *iov, size_t cnt, char *buf,
 			 * the remaining IOV count. This is so we can reserve at
 			 * least a length of 1 for every IOV.
 			 */
-			weight = (rand() % (len - (cnt - i))) + 1;
+			weight = rand() % ((len - cnt) + 1);
```

Comment:
I think this still allows weight to be zero.  The +1 needs to be outside, since rand() can return 0.